zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -1150,6 +1153,248 @@ benchmarks! {
+        let (caller, market_id) = create_market_common::<T>(
+            MarketCreation::Permissionless,
+            MarketType::Categorical(T::MaxCategories::get()),
+            ScoringRule::CPMM,
+            Some(MarketPeriod::Timestamp(range_start..old_range_end)),
+        )?;
+
+        for i in 0..o {
+            MarketIdsPerCloseTimeFrame::<T>::try_mutate(
+                Pallet::<T>::calculate_time_frame_of_moment(old_range_end),
+                |ids| ids.try_push(i.into()),
+            ).unwrap();
+        }
+
+        let now_time = <zrml_market_commons::Pallet::<T>>::now();
+        let new_range_end: MomentOf<T> = now_time + CloseEarlyProtectionTimeFramePeriod::get();
+
+        for i in 0..n {
+            MarketIdsPerCloseTimeFrame::<T>::try_mutate(
+                Pallet::<T>::calculate_time_frame_of_moment(new_range_end),
+                |ids| ids.try_push(i.into()),
+            ).unwrap();
+        }
+
+        Pallet::<T>::schedule_early_close(
+            RawOrigin::Signed(caller.clone()).into(),
+            market_id,
+        )?;
+
+        Pallet::<T>::dispute_early_close(
+            RawOrigin::Signed(caller.clone()).into(),
+            market_id,
+        )?;
+
+        let close_origin = T::CloseMarketEarlyOrigin::try_successful_origin().unwrap();
+        let call = Call::<T>::schedule_early_close { market_id };
+    }: { call.dispatch_bypass_filter(close_origin)? }
+
+    schedule_early_close_as_market_creator {
+        let o in 0..63;
+        let n in 0..63;
+
+        let range_start: MomentOf<T> = 0u64.saturated_into();
+        let old_range_end: MomentOf<T> = 100_000_000_000u64.saturated_into();
+        let (caller, market_id) = create_market_common::<T>(
+            MarketCreation::Permissionless,
+            MarketType::Categorical(T::MaxCategories::get()),
+            ScoringRule::CPMM,
+            Some(MarketPeriod::Timestamp(range_start..old_range_end)),
+        )?;
+
+        let market = <zrml_market_commons::Pallet::<T>>::market(&market_id)?;
+        let market_creator = market.creator.clone();
+
+        for i in 0..o {
+            MarketIdsPerCloseTimeFrame::<T>::try_mutate(
+                Pallet::<T>::calculate_time_frame_of_moment(old_range_end),
+                |ids| ids.try_push(i.into()),
+            ).unwrap();
+        }
+
+        let now_time = <zrml_market_commons::Pallet::<T>>::now();
+        let new_range_end: MomentOf<T> = now_time + CloseEarlyTimeFramePeriod::get();
+
+        for i in 0..n {
+            MarketIdsPerCloseTimeFrame::<T>::try_mutate(
+                Pallet::<T>::calculate_time_frame_of_moment(new_range_end),
+                |ids| ids.try_push(i.into()),
+            ).unwrap();
+        }
+
+        let origin = RawOrigin::Signed(market_creator).into();
+        let call = Call::<T>::schedule_early_close { market_id };
+    }: { call.dispatch_bypass_filter(origin)? }
```

Comment:
Missing `verify` function.

"No path dependency" is no argument. The extrinsic will silently fail if something goes wrong and you will never know and benchmark executing the error branch. Look at some standard libraries, they have no path dependence in many cases and always `verify` the success of the call:

- [pallet-balances](https://github.com/paritytech/polkadot-sdk/blob/39c04fdd9622792ba8478b1c1c300417943a034b/substrate/frame/balances/src/benchmarking.rs)
- [pallet-assets](https://github.com/paritytech/polkadot-sdk/blob/39c04fdd9622792ba8478b1c1c300417943a034b/substrate/frame/assets/src/benchmarking.rs)
- [pallet-utility](https://github.com/paritytech/polkadot-sdk/blob/39c04fdd9622792ba8478b1c1c300417943a034b/substrate/frame/utility/src/benchmarking.rs)
- etc.

Reply:
This is nonsense.